pulp_puppet_plugins/pulp_puppet/forge/api.py

```diff
@@ -98,9 +124,117 @@ def _get_module_name():
-        if MODULE_PATTERN.match(module_name):
-            return module_name
+        match = MODULE_PATTERN.match(module_name)
+        if match:
+            normalized_name = u'%s/%s' % (match.group(1), match.group(3))
+            return normalized_name
+
 
+class ReleasesPost36(Releases):
+
+    @staticmethod
+    def _format_query_string(base_url, module_name, module_version, offset, limit):
+        """
+        Build the query string to be used for creating
+
+        :param base_url: The context root to sue when generating a releases query.
+        :type base_url: str
+        :param module_name: The module name to add to the query string
+        :type module_name: str
+        :param module_version: The version of the module to encode in the query string
+        :type module_version: str
+        :param offset: The offset to encode for pagination
+        :type offset: int
+        :param limit: The max number of items to show on a page
+        :type limit: int
+        :return: The encoded URL for the specified query arguments
+        :rtype: str
+        """
+        query_args = {'module': module_name,
+                      'offset': offset,
+                      'limit': limit}
+        if module_version:
+            query_args['version'] = module_version
+
+        return '%s?%s' % (base_url, urllib.urlencode(query_args))
+
+    def get_releases(self, *args, **kwargs):
+        """
+        Get the list of matching releases
+
+        :return: The matching modules
+        :rtype: dict
+        """
+        return releases.view(*args, recurse_deps=False, view_all_matching=True, **kwargs)
+
+    def format_results(self, data):
+        """
+        Format the results and begin streaming out to the caller for the v3 API
+
+        :param data: The module data to stream back to the caller
+        :type data: dict
+        :return: the body of what should be streamed out to the caller
+        :rtype: str
+        """
+        web.header('Content-Type', 'application/json')
+        limit = int(web.input().get('limit', 20))
+        current_offset = int(web.input().get('offset', 0))
+        module_name = web.input().get('module', '')
+        module_version = web.input().get('version', None)
+        base_url_string = '/v3%s' % web.ctx.path
+
+        first_path = self._format_query_string(base_url_string, module_name, module_version,
+                                               0, limit)
+        current_path = self._format_query_string(base_url_string, module_name, module_version,
+                                                 current_offset, limit)
+        if current_offset > 0:
+            previous_path = self._format_query_string(base_url_string, module_name, module_version,
+                                                      current_offset - limit, limit)
+        else:
+            previous_path = None
+
+        formatted_results = {
+            "pagination": {
```

Comment:
From here down, there are a couple of cases of double quotes instead of single. Neither is wrong, but for consistency, it would be better to stick with single.